FWCore/Services/src/SiteLocalConfigService.cc

```diff
@@ -185,7 +185,9 @@ namespace edm {
 
     std::filesystem::path const SiteLocalConfigService::storageDescriptionPath(
         edm::CatalogAttributes const &aDataCatalog) const {
-      std::string siteconfig_path = std::string(std::getenv("SITECONFIG_PATH"));
+      char *tmp = std::getenv("SITECONFIG_PATH");
+      assert(tmp);  // Silence LLVM analyzer warning
```

Comment:
I'd prefer an exception be thrown if the environment variable were missing.

Reply:
Is it even a possible scenario (running without `SITECONFIG_PATH` set)?

Reply:
From CMSSW standpoint it is possible to get here without `SITECONFIG_PATH` being set. I agree with Chris that an exception would be better.